pom.xml

```diff
@@ -40,6 +40,11 @@
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>plain-credentials</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>token-macro</artifactId>
+            <version>2.15</version>
```

Comment:
this plugin is using Jenkins BOM. Any reason for pinning a concrete version?
Besides, could this dependency be optional?

Reply:
> could this dependency be optional

Or removed. I think you can just check `build`’s environment variables, unless it is null (which it may be).

Reply:
I tried to use environment variables in my previous versions, but unfortunately it's not that easy. The git environment variables seem to be not accessible from the rest of the code. This is a known issue and in many threads even the plugin owners recommends the use of token macros.
I agree we should remove the version.